From a31550a73d795753a6f8272f4eb967e265a093f0 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 16 Aug 2018 21:48:41 +0300
Subject: [PATCH] Makefile [gaudi] add new goal

Add gaudi-onix-tools-install goal and update tcl to load gaudi bit file
---

diff --git a/Makefile b/Makefile
index ae074a1..9d5d94b 100644
--- a/Makefile
+++ b/Makefile
@@ -179,8 +179,9 @@
 		$(MAKE) host-uboot-test-hooks-rebuild
 
 dali-onix-tools-install: DALI_INSTALL_LAB_IP ?= monster4-nuc
-.PHONY: dali-onix-tools-install
-dali-onix-tools-install:
+gaudi-onix-tools-install: DALI_INSTALL_LAB_IP ?= monster4-nuc
+.PHONY: dali-onix-tools-install gaudi-onix-tools-install
+dali-onix-tools-install gaudi-onix-tools-install:
 	-cd board/habanalabs/common/ &&					\
 	sshpass -p123456 scp -r remote_power ip_pdu t32 vivado_lab	\
 	root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
diff --git a/board/habanalabs/common/vivado_lab/program-dspg.tcl b/board/habanalabs/common/vivado_lab/program-dspg.tcl
index f8293a7..422a095 100644
--- a/board/habanalabs/common/vivado_lab/program-dspg.tcl
+++ b/board/habanalabs/common/vivado_lab/program-dspg.tcl
@@ -1,10 +1,16 @@
 open_hw
 connect_hw_server
 open_hw_target
-set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_dspg_070618.bit} [get_hw_devices xcvu440_0]
+# DALI:
+#set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_dspg_070618.bit} [get_hw_devices xcvu440_0]
+# GAUDI:
+set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_130818.bit} [get_hw_devices xcvu440_0]
 current_hw_device [get_hw_devices xcvu440_0]
 refresh_hw_device [lindex [get_hw_devices xcvu440_0] 0]
-set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_dspg_070618.bit} [get_hw_devices xcvu440_0]
+# DALI:
+# set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_dspg_070618.bit} [get_hw_devices xcvu440_0]
+# GAUDI:
+set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_130818.bit} [get_hw_devices xcvu440_0]
 program_hw_devices [get_hw_devices xcvu440_0]
 refresh_hw_device [lindex [get_hw_devices xcvu440_0] 0]
 exit
diff --git a/board/habanalabs/common/vivado_lab/viv.sh b/board/habanalabs/common/vivado_lab/viv.sh
index f88fb61..d950b49 100755
--- a/board/habanalabs/common/vivado_lab/viv.sh
+++ b/board/habanalabs/common/vivado_lab/viv.sh
@@ -38,3 +38,5 @@
 fi
 sshpass -p123456 scp ${VIV_WORK}/${VIV_PROG_TCL} root@${HOST_NAME}:${VIV_WORK}/
 sshpass -p123456 ssh root@${HOST_NAME} "cd ${VIV_WORK} && ${VIV_PROG_CMD} > /dev/null"
+sshpass -p123456 ssh root@${HOST_NAME} "reboot"
+sleep 60
